git-branchless/src/commands/reword.rs

```diff
@@ -90,6 +93,43 @@ pub fn reword(
         }
     };
 
+    let messages = match messages {
+        InitialCommitMessages::Discard | InitialCommitMessages::Messages(_) => messages,
+        InitialCommitMessages::FixUp(revset) => {
+            let commits_to_fixup =
+                resolve_commits_from_hashes(&repo, &mut dag, effects, vec![revset.clone()])?
+                    .unwrap_or_default();
+            let commit_to_fixup = match commits_to_fixup.as_slice() {
+                [commit_to_fixup] => {
+                    if commits.iter().any(|commit| {
+                        !dag.query()
+                            .is_ancestor(commit_to_fixup.get_oid().into(), commit.get_oid().into())
+                            .expect("Failed checking ancestry of commit.")
+                    }) {
+                        writeln!(
+                            effects.get_error_stream(),
+                            "The commit supplied to --fixup must be an ancestor of all commits being reworded.\nAborting.",
+                        )?;
+                        return Ok(ExitCode(1));
+                    }
+                    commit_to_fixup
+                }
+                commits => {
+                    writeln!(
+                        effects.get_error_stream(),
+                        "--fixup expects exactly 1 commit, but '{}' evaluated to {}.\nAborting.",
+                        revset.0,
```

Comment:
Feel free to implement `Display` for `Revset` instead of accessing `.0`.

Reply:
> I think you could write this as `dag.query().common_ancestors(...`

Oh, yes ... much nicer, thank you!

> Feel free to implement `Display` for `Revset`

Done!